lighthouse-core/gather/driver.js

```diff
@@ -881,6 +888,11 @@ class Driver {
       return function() {
         log.verbose('Driver', 'loadEventFired and network considered idle');
       };
+    }).catch(err => {
+      // Throw the error in the cleanupFn so we still cleanup all our handlers.
```

Comment:
this isn't getting easier to follow... :P

Reply:
no, it's not, but it's very well tested now and ready for refactoring :)